src/tdastro/astro_utils/passbands.py

```diff
@@ -2,257 +2,246 @@
+    transmission_table : np.ndarray
+        A 2D array where the first col is wavelengths (Angstrom) and the second col is transmission values.
+    normalized_transmission_table : np.ndarray
+        A 2D array of wavelengths and normalized transmissions.
+    """
 
-            logging.info(f"Downloaded: {band_name}-band transmission table.")
-            return True
-        except HTTPError as e:
-            logging.error(f"HTTP error occurred when downloading transmission table for {band_name}: {e}")
-            return False
-        except URLError as e:
-            logging.error(f"URL error occurred when downloading transmission table for {band_name}: {e}")
-            return False
+    def __init__(self, survey, label, table_path=None, table_url=""):
+        self.label = label
+        self.survey = survey
+        self.full_name = f"{survey}_{label}"
+        self.table_path = table_path
+        self.table_url = table_url
+        self.transmission_table = self._load_transmission_table()
+        self.normalized_transmission_table = self._normalize_transmission_table()
 
-    def _load_transmission_table(self, band_name: str, file_path: str) -> bool:
-        """Load a transmission table from a file.
+    def _load_transmission_table(self) -> np.ndarray:
+        """Load a transmission table from a file or URL.
 
-        Note
-        ----
-        This is a temporary solution to be replaced when we implement Pooch throughout the project.
+        Returns
+        -------
+        np.ndarray
+            A 2D array where the first column is wavelengths (Angstrom) and the second column is transmission
+            values.
+        """
+        if self.table_path is None:
+            self.table_path = os.path.join(
+                os.path.dirname(__file__), f"passbands/{self.survey}/{self.label}.dat"
+            )
+            os.makedirs(os.path.dirname(self.table_path), exist_ok=True)
+        if not os.path.exists(self.table_path):
+            self._download_transmission_table()
+        return np.loadtxt(self.table_path)
 
-        Parameters
-        ----------
-        band_name : str
-            Name of the band for which the transmission table is to be loaded.
-        file_path : str
-            Path to the file containing the transmission table.
+    def _download_transmission_table(self) -> bool:
+        """Download a transmission table from a URL.
 
         Returns
         -------
         bool
-            True if the transmission table is successfully loaded, False otherwise.
+            True if the download was successful, False otherwise.
         """
+        if self.table_url == "":
+            if self.survey == "LSST":
+                # TODO: This area will be changed with incoming Pooch data manager PR :)
+                # Consider: https://github.com/lsst/throughputs/blob/main/baseline/filter_g.dat
+                # Or check with Mi's link (unless that was above)
```

Comment:
I think we want to use this one: https://github.com/lsst/throughputs/blob/main/baseline/total_g.dat